clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -767,9 +775,38 @@ private void handleListOffsetResponse(Map<TopicPartition, Long> timestampsToSear
                     return null;
                 }
 
+                Set<Long> abortedPids = new HashSet<>();
+
+                /* When in READ_COMMITTED mode, we need to do the following for each incoming entry:
+                 *   0. Check whether the pid is in the 'abortedPids' set && the entry does not include an abort marker.
+                 *      If so, skip the entry.
+                 *   1. If the pid is in aborted pids and the entry contains an abort marker, remove the pid from
+                 *      aborted pids and skip the entry.
+                 *   2. Check lowest offset entry in the abort index. If the PID of the current entry matches the
+                 *      pid of the abort index entry, and the incoming offset is at least the abort index offset,
+                 *      this means that the entry has been aborted. Add the pid to the aborted pids set, and remove
+                 *      the entry from the abort index.
+                 */
+                PriorityQueue<FetchResponse.AbortedTransaction> abortedTransactions = getAbortedTransactions(partition);
+
                 List<ConsumerRecord<K, V>> parsed = new ArrayList<>();
                 boolean skippedRecords = false;
                 for (RecordBatch batch : partition.records.batches()) {
+                    if (isolationLevel == IsolationLevel.READ_COMMITTED) {
+                        FetchResponse.AbortedTransaction nextAbortedTransaction = abortedTransactions.peek();
+                        if (abortedPids.contains(batch.pid())
```

Comment:
Could we just follow the above comments and follow that as two separate conditions (i.e. we can merge 1) and 2) together as a nested)? Personally I feel it is clearer to duplicate the logic of setting `skippedRecords = true`, and also we can avoid calling `RecordBatch#baseOffset()` and RecordBatch#pid()`.